frontend_tests/node_tests/message_edit.js

```diff
@@ -0,0 +1,57 @@
+set_global('document', null);
+set_global('page_params', {});
+
+add_dependencies({
+    XDate: 'third/xdate/xdate.dev.js',
+});
+
+var message_edit = require('js/message_edit.js');
+var get_editability = message_edit.get_editability;
+var editability_types = message_edit.editability_types;
+
+(function test_get_editability() {
+    assert.equal(get_editability(null), editability_types.NO);
+    assert.equal(get_editability({
+        sent_by_me: false,
+    }), editability_types.NO);
+
+    assert.equal(get_editability({
+        sent_by_me: true,
+        failed_request: true,
+    }), editability_types.FULL);
+
+    assert.equal(get_editability({
+        sent_by_me: true,
+        local_id: 25,
+    }), editability_types.NO);
```

Comment:
you should add comments explaining the meaning of the various cases, so that one can understand why these are the correct result.  E.g. for this one, "Locally echoed messages are not editable, since the message hasn't actually finished being sent yet".

Otherwise, looks good.